Assets/LoggingManager/LoggingManager.cs

```diff
@@ -126,220 +95,147 @@ public void SaveLog(string collectionLabel)
-                }
+                AddMetaCollectionToList();
             }
-            int count = collections[collectionLabel].count;
-            if (logMode == LogMode.Append)
+            else
             {
-                if (collections[collectionLabel].log[pair.Key].ContainsKey(count))
-                {
-                    collections[collectionLabel].count++;
-                    count = collections[collectionLabel].count;
-                }
+                logsList.Add(collectionLabel, new LogStore(collectionLabel, email, sessionID, logStringOverTime));
             }
-
-            collections[collectionLabel].log["Timestamp"][count] = GetTimeStamp();
-            collections[collectionLabel].log["Framecount"][count] = GetFrameCount();
-            collections[collectionLabel].log["SessionID"][count] = sessionID;
-            collections[collectionLabel].log["Email"][count] = email;
-            collections[collectionLabel].log[pair.Key][count] = pair.Value;
         }
-    }
 
-    public void Log(string collectionLabel, string columnLabel, object value, LogMode logMode = LogMode.Append)
-    {
-        if (!collections.ContainsKey(collectionLabel))
+        LogStore logStore = logsList[collectionLabel];
+
+        foreach (KeyValuePair<string, object> pair in logData)
         {
-            collections.Add(collectionLabel, new LogCollection());
-            collections[collectionLabel].label = collectionLabel;
+            logStore.Add(pair.Key, pair.Value);
         }
+    }
 
-        if (!collections[collectionLabel].log.ContainsKey(columnLabel))
-        {
-            collections[collectionLabel].log.Add(columnLabel, new Dictionary<int, object>());
+    private void AddMetaCollectionToList()
+    {
+        LogStore metaLog = new LogStore("Meta", email, sessionID, logStringOverTime, LogType.OneRowOverwrite);
+        logsList.Add("Meta", metaLog);
+        metaLog.Add("SessionID", sessionID);
+        metaLog.Add("DeviceID", deviceID);
+    }
 
-            if (!collections[collectionLabel].log.ContainsKey("Timestamp"))
-            {
-                collections[collectionLabel].log["Timestamp"] = new Dictionary<int, object>();
-            }
-            if (!collections[collectionLabel].log.ContainsKey("Framecount"))
-            {
-                collections[collectionLabel].log["Framecount"] = new Dictionary<int, object>();
-            }
-            if (!collections[collectionLabel].log.ContainsKey("SessionID"))
-            {
-                collections[collectionLabel].log["SessionID"] = new Dictionary<int, object>();
-            }
-            if (!collections[collectionLabel].log.ContainsKey("Email"))
-            {
-                collections[collectionLabel].log["Email"] = new Dictionary<int, object>();
-            }
-        }
+    public void TerminateLogRow(string collectionLabel)
+    {
+        logsList[collectionLabel].EndRow();
+    }
```

Comment:
Why do we need this function in LoggingManager? In the old logging manager, the behavior was the following:

- If the user wants to log a single value, the user calls `Log(string collectionLabel, string columnLabel, object value)`.
- If the user wants to log multiple values, the user calls `Log(string collectionLabel, Dictionary<string, object> logData)`.

And this way LoggingManager could always infer when a line will end - the convention was that calling `Log()` always corresponds to 1 row in a  CSV file - while it is less powerful, i think it makes it simple to understand by design.  It eliminates extra work for the user (data always appear on a new row). In addition it is more accurate because the timestamp with milliseconds will always correspond very closely to the logged event.

Reply:
I didn't know the old LoggingManager was designed like this.
What I did here is a system which can detect when a new line is added.
For instance, let's say we want to log 3 columns : positionX, positionY and positionZ.

* We can log using ```Log(string collectionLabel, string columnLabel, object value)``` 
    * ```c# 
	  loggingManager.Log("Label","positionX",100);
	  loggingManager.Log("Label","positionY",200);
	  loggingManager.Log("Label","positionZ",100);
	   ```
* Or ```Log(string collectionLabel, Dictionary<string, object> logData)``` 
    * ```c# 
	  loggingManager.Log("Label",new Dictionary<string,object>(){
		  {"positionX",100}
		  {"positionY",200}
		  {"positionZ",100}
	  };
	   ```
In our example, our logs would be like this : 
| positionX | positionY | positionZ |
|--|--|--|
| 100 | 200 | 100 |

Now, if we add a new row having the "positionX" as column label, the program will detect that the column has already been logged for this row and will automatically create a new row.

So after this line of code: 
```c# 
loggingManager.Log("Label","positionX",150);
```
The logs will be like this: 
| positionX | positionY | positionZ |
|--|--|--|
| 100 | 200 | 100 |
| 150 |  |  |

The user can also call the ```TerminateLogRow()``` method manually if he wants to manually end a line at a specific point.

There is also the second feature that adds a "NULL" value to each previous rows when adding a new header.

With these two features, we are sure that each row will have the same number of columns without the user having to worry about it (see [this comment](https://github.com/med-material/ArduinoLogger/pull/46#issuecomment-988644241)).

So this behavior is not very complicated to understand by the users and make the code very easy to write : the user only has to call the ```Log()``` method when he wants to log something and that's it, the program manages everything, but if he needs to do something more specific, he can always end a row using the ```TerminateLogRow()``` method. 


After these explanations, do you still think it's better to end a line after each call of ```Log()``` ?
We could also put a boolean in the LoggingManager that, if true, will end the line after each call to ```Log()``` (it would be set to true by default).

Reply:
@Vincent-SD  The main issue with this logic, is that it makes reading `Timestamp`, `Framecount` etc. for each row problematic  - there is no longer a forced correlation between the reported timestamp and the time which the values were created (it could become arbitrary). It means, that if users are not aware of this, they can run into a lot of trouble in data analysis. 

This is the main reason that the LoggingManager was designed to always terminate the line, after `Log()` is called  - and to force a `Timestamp` on every single row. This behavior was not a limitation/simplification, but a feature.

Unfortunately it is very important to maintain this correlation for any type of temporal data analysis, so I don't think we should put a boolean in `Log()` for this.

Reply:
Okay, thanks for the explanations. I'll do this.